### PR TITLE
CANDLEPIN-547: i18n job needs to set Java17 for main branch

### PR DESCRIPTION
**Pull request by bakajstep**
- Set Java version from 11 to 17 in internationalization job
- Fix build docker image

### CODE REVIEW

File: src/resources/candlepin-internationalization.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ for GIT_BRANCH in main candlepin-4.1-HOTFIX candlepin-4.2-HOTFIX; do
   git checkout $GIT_BRANCH
   evalrc $? "Checkout branch : $GIT_BRANCH was not successful."
 
-  JAVA_VERSION=11
+  JAVA_VERSION=17
```
Comment (nikosmoum) [inline src/resources/candlepin-internationalization.sh]: we cannot just set java 17 for all branches. We want java 17 to be set only for the main branch, and Java 11 for older branches

Comment (nikosmoum) [reply src/resources/candlepin-internationalization.sh]: You can see here the code I removed, was doing this switch between 8 and 11: https://github.com/candlepin/candlepin-jobs/commit/013958b23ef3f014f4bc263503b70e0ce5d96955

**Review by nikosmoum — CHANGES_REQUESTED** [review]

File: src/resources/candlepin-internationalization.sh (hunk 1 above)
Comment (bakajstep) [reply src/resources/candlepin-internationalization.sh]: done

File: src/resources/candlepin-internationalization.sh [hunk 2]
```diff
@@ -19,17 +19,24 @@ get_simplified_version() {
...
+for key in "${!candlepin_java[@]}"; do
 
   #Clean the project
   ./gradlew clean
-  evalrc $? "Project cleaning was not successful for branch $GIT_BRANCH."
+  evalrc $? "Project cleaning was not successful for branch $key."
 
   #Checkout the branch
-  git checkout $GIT_BRANCH
-  evalrc $? "Checkout branch : $GIT_BRANCH was not successful."
```
Comment (bakajstep) [inline src/resources/candlepin-internationalization.sh:30]: git checkout return name of the switched branch. So this function always evaluate it as a error.

**Review by nikosmoum — APPROVED** [review]

**Pull request merged**